eng/scripts/automation/Invoke-GenerateAndBuild.ps1

```diff
@@ -41,27 +42,59 @@ if ($autorestConfig -ne "") {
     }
 }
 
-Write-Host "swaggerDir:$swaggerDir, readmeFile:$readmeFile"
+$generatedSDKPackages = New-Object 'Collections.Generic.List[System.Object]'
 
 # $service, $serviceType = Get-ResourceProviderFromReadme $readmeFile
 $sdkPath =  (Join-Path $PSScriptRoot .. .. ..)
 $sdkPath = Resolve-Path $sdkPath
 $sdkPath = $sdkPath -replace "\\", "/"
 
-$readme = ""
-if ($commitid -ne "") {
-  if ($repoHttpsUrl -ne "") {
-    $readme = "$repoHttpsUrl/blob/$commitid/$readmeFile"
+if ($readmeFile) {
+  Write-Host "swaggerDir:$swaggerDir, readmeFile:$readmeFile"
+
+  $readme = ""
+  if ($commitid -ne "") {
+    if ($repoHttpsUrl -ne "") {
+      $readme = "$repoHttpsUrl/blob/$commitid/$readmeFile"
+    } else {
+      $readme = "https://github.com/$org/azure-rest-api-specs/blob/$commitid/$readmeFile"
```

Comment:
Is this not intended to support cadl in the private repo i.e., Azure/azure-rest-api-specs-pr?

Also, what other org is there? Seems like an odd variable to have, or is this meant to also support forks?

Reply:
Current `cadl compile` can only accept file-path, it cannot load cadl file via url. So we need to clone the swagger repo locally .